### DISCUSSION TITLE
Is there a way to save the instances generated by ecole.instance as mps file supported by PySCIPOpt

### DISCUSSION THREAD

**Discussion by thupchnsky** [ds4dm/ecole#171] [Questions] [answered]
Hi,

I am trying to save the instances generated by ecole.instance.SetCoverGenerator as mps files beacuse I need them to be used by different algorithms for fair comparison. However, I dont find the direct write out method for these instances. I know that PySCIPOpt has method like `model.writeProblem("xxx.mps")`. So I am wondering how should I convert the instance to pyscipopt.scip.model so I can save the files? Thanks!

**Comment by AntoinePrv** [comment, marked as answer]
Hello,

`Model.write_problem` will forward to [`SCIPwriteOrigProblem`](https://www.scipopt.org/doc-7.0.2/html/group__GlobalProblemMethods.php#gaf882d322c96b7edf4ae4240e6f35d4b5), and the extension is detected by SCIP, so you can
```python
model.write_problem("my-instance.mps")
```
Or 
```python
model.write_problem("my-instance.lp")
```
...

**Reply by thupchnsky** [reply]
I find out that `generate_instance` class returns a scip.Model. So should I do something like this?
```python
instances = ecole.instance.SetCoverGenerator(n_rows=500, n_cols=1000, density=0.05)
model = next(instances)
model.write_poblem("my-instance.mps")
```
I assume this will do the job?

**Reply by AntoinePrv** [reply]
Yes, you could also do
```python
model.as_pyscipopt().writeProblem("my-instance.mps")
```
If you wanted to use the PySCIPOpt method (which currently has a few more options).

**Reply by thupchnsky** [reply]
Well it seems that if I am using `model.write_poblem("my-instance.mps")`, it complains about `'ecole.core.scip.Model' object has no attribute 'write_poblem'`. 

But `model.as_pyscipopt().writeProblem("my-instance.mps")` works.

**Reply by AntoinePrv** [reply]
> Well it seems that if I am using model.write_poblem("my-instance.mps"), it complains about 'ecole.core.scip.Model' object has no attribute 'write_poblem'.

It works for me. I don't remember when we added it. Are you running the latest version of Ecole (`v0.6.0`)?

**Reply by thupchnsky** [reply]
I am using Python 3.8.8 with ecole 0.6.0 and pyscipopt 3.1.2. Not sure what the problem would be... I am just using the following code
```python
import ecole
instances = ecole.instance.SetCoverGenerator(n_rows=500, n_cols=1000, density=0.05)
model = next(instances)
model.write_poblem("my-instance.mps")
```

**Reply by AntoinePrv** [reply]
Haha, I got the same problem in the end running the code sample! I made a typo:
`write_poblem` > `write_problem`

**Reply by AntoinePrv** [reply]
You can find the methods we currently have on `ecole.scip.Model` in the [doc](https://doc.ecole.ai/master/reference/scip-interface.html).

We don't aim to bind everything, only highly used methods, or methods not offered in PySCIPOpt.

**Reply by thupchnsky** [reply]
Thanks for the help! That was really an embrassing typo lol...
